app.py

```
@@ -52,19 +65,20 @@ def rename_file_if_needed(source_path, target_path):
 def analyze_file(file_path, demucs_model):
+    # Perform analysis
     global progress, audio_file_url, tempo
     try:
         progress = 20  # Begin analysis
         y, sr = librosa.load(file_path, sr=None)
         tempo, _ = librosa.beat.beat_track(y=y, sr=sr)
-        tempo = float(tempo) if not isinstance(tempo, np.ndarray) else tempo.item()
+        tempo = round(float(tempo) if not isinstance(tempo, np.ndarray) else tempo.item(), 2)
         progress = 40  # Audio file loaded successfully
 
         subprocess.run(['demucs', '-n', demucs_model, file_path], check=True)
```

Ruff (0.8.0)

`subprocess` call: check for execution of untrusted input

(S603)